tools/make/golang.mk

```diff
@@ -22,12 +22,14 @@ go.build.%:
 
 # Build the envoy-gateway binaries in the hosted platforms.
 .PHONY: go.build
-go.build:  $(addprefix go.build., $(addprefix $(PLATFORM)., $(BINS)))
+go.build:
+	@$(MAKE) $(addprefix go.build., $(addprefix $(PLATFORM)., $(BINS)))
```

Comment:
What's the point of this change? In the other places that do `$(MAKE)` like this, I'd have made the reverse change.

(See also: Peter Miller's famous "Recursive Make Considered Harmful" article)

Reply:
@LukeShu thanks for the review, I added that extra `make` because I was unable to work around it, here's the error
```
diff --git a/tools/make/golang.mk b/tools/make/golang.mk
index 7600d0a..9c7f1c3 100644
--- a/tools/make/golang.mk
+++ b/tools/make/golang.mk
@@ -22,8 +22,7 @@ go.build.%:
 
 # Build the envoy-gateway binaries in the hosted platforms.
 .PHONY: go.build
-go.build:
-       @$(MAKE) $(addprefix go.build., $(addprefix $(PLATFORM)., $(BINS)))
+go.build: $(addprefix go.build., $(addprefix $(PLATFORM)., $(BINS)))
 
 # Build the envoy-gateway binaries in multi platforms
 # It will build the linux/amd64, linux/arm64, darwin/amd64, darwin/arm64 binaries out.
🐳 ~/go-workspace/src/github.com/envoyproxy/gateway$ make build
make[3]: Nothing to be done for `go.build'.
```
hopefully some `make` magician can help iron this out, but for the time being shouldnt affect correctness